test/DynamoCoreTests/CustomNodeWorkspaceOpening.cs

```diff
@@ -38,6 +38,21 @@ public void CanOpenWorkspaceWithMissingCustomNodeThenFixByOpeningNeededCustomNod
         }
 
         [Test]
+        public void CanLoadReadOnlyNode()
+        {
+            // Open a read-only custom node
+            OpenTestFile(@"core\CustomNodes", "add_Read_only.dyf");
```

Comment:
I removed write privilege to this file for all user. Sure I can add a check here.